appmode/server_extension.py

```diff
@@ -1,17 +1,29 @@
 # -*- coding: utf-8 -*-
 
 import os
+import inspect
 import itertools
 from notebook.utils import url_path_join
 from notebook.base.handlers import IPythonHandler, FilesRedirectHandler, path_regex
 import notebook.notebook.handlers as orig_handler
 import notebook
-import collections.abc
 from tornado import web
 from traitlets.config import LoggingConfigurable
 from traitlets import Bool, Unicode
 
 
+async def ensure_async(obj):
```

Comment:
The name is a bit confusing with the `asyncio.ensure_future`. I would suggest call the function `def await_if_awaitable(obj)`.

Reply:
I've renamed it to await_if_awaitable. The name ensure_async was what jupyter_core used for the function, but I don't have any strong opinions on it.

Reply:
Okay, then I think they are all fine, since you already changed them, I won't ask you to change them back :smile_cat: . Thanks!

Reply:
No worries, easy enough to change back if you think the other name was better, or you think of another one. Thanks for the quick reviews!